cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/OperationRunnable.java

```diff
@@ -43,6 +43,7 @@ public void run() {
           ((Pending) steps.get(0)).done();
         }
         _future.complete(getResult());
+        _future.setFinishTime(System.nanoTime());
```

Comment:
Can we clarify whether we are sure that the future is complete here -- i.e. we don't set the finish time prematurely?

Reply:
I think so, this is the only place where we set the finish time and we set it at the end of run method, which guarantees that it is set only when this method finishes without error. 

Since all the async endpoint task is encapsulated in an subclass of OperationRunnable by overriding its getResult(), which is before the setFinishTime(), we are sure the system time we get at setFinishTime is after all the operation is done.

There is one thing to notice, the endpoint execution time we measure here does not include the execution time of executor for add/remove/demote/rebalance with dryrun=false. this is because the tasks submitted to executor asynchronously. we return the response and close the session once the task is submitted to executor.